src/graphql/schemas/inputs/signatureRequestInput.ts

```diff
@@ -0,0 +1,34 @@
+import { Field, InputType } from "type-graphql";
+
+import { SignatureRequest } from "../typeDefs/signatureRequestTypeDefs.js";
+
+import type { WhereOptions } from "./whereOptions.js";
+import {
+  NumberSearchOptions,
+  SignatureRequestPurposeSearchOptions,
+  SignatureRequestStatusSearchOptions,
+  StringSearchOptions,
+} from "./searchOptions.js";
+
+@InputType()
+export class BasicSignatureRequestWhereInput
+  implements WhereOptions<SignatureRequest>
+{
+  @Field(() => StringSearchOptions, { nullable: true })
+  safe_address?: StringSearchOptions;
+
+  @Field(() => StringSearchOptions, { nullable: true })
+  message_hash?: StringSearchOptions;
+
+  @Field(() => StringSearchOptions, { nullable: true })
+  created_at?: StringSearchOptions;
```

Comment:
My preference would be a single format for datetime. We basically stay as close to the on-chain data as possible, which would make this a timestamp in second. The datatime used in the other tables also shouldn't be there. It's terrible DX to need to figure out which timestamp you're getting.

Reply:
See [this example query](https://staging-api.hypercerts.org/v1/graphql?query=%7B%0A++attestations%28first%3A+1%29+%7B%0A++++data+%7B%0A++++++last_update_block_timestamp%0A++++++last_update_block_number%0A++++++creation_block_number%0A++++++creation_block_timestamp%0A++++%7D%0A++%7D%0A%7D).

The data is stored as `numeric(78,0)` because Postgres, and the numbers are parsed to string representation after fetching because JSON.

Reply:
Just for the sake of argument, JSON accomodates Unix timestamps (signed 32 bit) as numbers. You don't need to stringify them.